File: DistributeCandies/DisCan.py
class Solution:
    def distributeCandies(self, candyType: List[int]) -> int:
        n = len(candyType)
        maxim = n // 2
        m = set(candyType)
        i = len(m)
        if i > maxim:
            return maxim
        else: return i


class Solution:
    def distributeCandies(self, candyType: List[int]) -> int:
        return min(len(candyType) // 2, len(set(candyType)))

File: DistributeCandies/DisCan.js
// Метод доступа size возвращает количество (уникальных) элементов в объекте Set

/**
 * @param {number[]} candyType
 * @return {number}
 */
 var distributeCandies = function(candyType) {
    let set = new Set(candyType);
    if (set.size >= candyType.length / 2) return candyType.length / 2;
    else return set.size;
};


/**
 * @param {number[]} candyType
 * @return {number}
 */
 var distributeCandies = function(candyType) {
    return Math.min(candyType.length/2,  new Set(candyType).size)
};